src/IO/Exporter/PointwiseInterpolator.cpp

```diff
@@ -131,18 +120,34 @@ void interpolate_to_points_impl(
                                               points.element_logical_coords);
       const size_t num_element_target_points =
           points.element_logical_coords.begin()->size();
-      if (interpolated_data.size() < num_element_target_points) {
-        interpolated_data.destructive_resize(num_element_target_points);
-      }
       for (size_t i = 0; i < tensor_data.size(); ++i) {
-        auto output_data =
-            gsl::make_span(interpolated_data.data(), num_element_target_points);
-        const auto input_data =
-            gsl::make_span(tensor_data[i].data() + offset, length);
-        interpolant.interpolate(make_not_null(&output_data), input_data);
-        for (size_t j = 0; j < num_element_target_points; ++j) {
-          (*result)[i][points.offsets[j]] = interpolated_data[j];
-          (*filled_data)[points.offsets[j]] = true;
+        const auto& component = tensor_data[i];
+        if (std::holds_alternative<DataVector>(component)) {
+          if (interpolated_data_double.size() < num_element_target_points) {
+            interpolated_data_double.destructive_resize(
+                num_element_target_points);
+          }
+          auto output_data = gsl::make_span(interpolated_data_double.data(),
+                                            num_element_target_points);
+          const auto input_data = gsl::make_span(
+              std::get<DataVector>(component).data() + offset, length);
+          interpolant.interpolate(make_not_null(&output_data), input_data);
+          for (size_t j = 0; j < num_element_target_points; ++j) {
+            (*result)[i][points.offsets[j]] = interpolated_data_double[j];
+            (*filled_data)[points.offsets[j]] = true;
+          }
+        } else {
+          interpolated_data_float.resize(num_element_target_points);
```

Comment:
Maybe only resize of size is below what you need? I'm not sure what guarantees there are on vector if you decrease size. Probably would be good to avoid reallocating inside a loop when possible

Reply:
Yea, I was unable to check cppreference because it was down while I was reviewing :(